core/chain-configs/src/genesis_config.rs

```diff
@@ -248,6 +249,8 @@ pub struct Genesis {
     /// so they should be processed in streaming fashion with for_each_record.
     #[serde(skip)]
     records_file: PathBuf,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub state_roots: Option<Vec<StateRoot>>,
```

Comment:
Maybe you are already working towards it, but this really looks like a time to finally rework Genesis contents as an enum. Because records, records_file, and state_roots are all mutually exclusive. I can pick up reworking Genesis into config + contents enum model in a separate issue, so that this PR is not that big, but you would have to rebase. Or you can do this in this PR, but somebody has to do it sometime, because this jenga tower is going to fall.

Reply:
Yes, please do, I can rebase, not an issue at all. Thanks!